sample/HttpTrigger-CSharp-CustomRoute/run.csx

```diff
@@ -0,0 +1,19 @@
+﻿#r "Newtonsoft.Json"
+
+using System;
+using System.Net;
+using System.Net.Http;
+using Microsoft.Azure.WebJobs.Host;
+
+public class ProductInfo
+{
+    public string Category { get; set; }
+    public string Id { get; set; }
+}
+
+public static ProductInfo Run(ProductInfo info, string category, string id, TraceWriter log)
+{
```

Comment:
I haven't gotten this far yet -- but are you setting `Category` and `Id` on the `info` parameter?